### PR TITLE
Log equality axiom evaluation

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Logger/DebugAppliedRule.hs [hunk 1]
```diff
@@ -113,9 +113,9 @@ debugAppliedRule rule =
 See also: 'parseDebugAppliedRuleOptions'
 
  -}
-data DebugAppliedRuleOptions =
+newtype DebugAppliedRuleOptions =
```
[the remark this answers is not in the document]
Comment (virgil-serbanuta) [reply kore/src/Kore/Logger/DebugAppliedRule.hs]: I would guess that it doesn't, but I don't know enough about how newtype works. @ttuegel, do you have an opinion?

File: kore/src/Kore/Logger/DebugAxiomEvaluation.hs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+{- | A log 'Entry' when a rule is applied.
+
+We will log the applied rule and its unification or matching condition.
+
+ -}
+data DebugAxiomEvaluation =
+    DebugAxiomEvaluation
+    { identifier :: !(Maybe AxiomIdentifier)
+    , secondaryIdentifier :: Maybe Text
```
[the remark this answers is not in the document]
Comment (virgil-serbanuta) [reply kore/src/Kore/Logger/DebugAxiomEvaluation.hs]: Fixed

File: kore/src/Kore/Logger/DebugAppliedRule.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ttuegel) [reply kore/src/Kore/Logger/DebugAppliedRule.hs]: It does impact the representation at runtime, but in a positive way. At runtime with `newtype` there is no heap object for the wrapper, that is: there is only a `Set Id` and there is no object in memory corresponding to a `DebugAppliedRuleOptions`. With `data`, there is a heap object for `DebugAppliedRuleOptions` containing a pointer to a `Set Id`.

File: kore/src/Kore/Logger/DebugAxiomEvaluation.hs [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+We will log the applied rule and its unification or matching condition.
+
+ -}
+data DebugAxiomEvaluation =
+    DebugAxiomEvaluation
+    { identifier :: !(Maybe AxiomIdentifier)
+    , secondaryIdentifier :: !(Maybe Text)
+    , state :: !AxiomEvaluationState
+    , severity :: !Severity
```
Comment (ttuegel) [inline kore/src/Kore/Logger/DebugAxiomEvaluation.hs]: Wouldn't the `Severity` always be `Debug`?

Comment (virgil-serbanuta) [reply kore/src/Kore/Logger/DebugAxiomEvaluation.hs]: Sometimes, i.e. for the symbols selected by `--debug-simplification-axiom`, it's "Info". That's how I interpreted our chat about this:

"We want to log all these things at the Debug level, but have options to print the entries for specific functions regardless of the --log-level selected by the user."

I guess that I should have used "Critical" to really print entries regardless of the `--log-level`, but I thought I'd try "Info" first. Should I log them as "Critical" instead?

Comment (ttuegel) [reply kore/src/Kore/Logger/DebugAxiomEvaluation.hs]: That's OK for now.

**Pull request merged**